src/lib.rs

```diff
@@ -163,53 +172,175 @@ pub mod pallet {
+        // ^
+        // |
+        // |
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        //
+        // DEPENDING ON PERFS, ONLY ONE OF THOSE CALL WILL BE KEPT
+        //
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        // |
+        // |
+        // v
+
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn submit_encoded_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			encoded_call: Vec<u8>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+			let nonce = Self::nonce_call(sindex);
+			let deposit = <BalanceOf<T>>::from(encoded_call.len() as u32)
+				.saturating_mul(T::PreimageByteDeposit::get());
+
+            T::Currency::reserve(&who, deposit)?;
+
+			let preimage = PreimageCall::<T::AccountId, BalanceOf<T>> {
+				data: encoded_call,
+				provider: who.clone(),
+				deposit,
+			};
+
+            Calls::<T>::insert(sindex, nonce, preimage);
+            Self::deposit_event(Event::<T>::CallSubmitted(supersig_id, nonce, who));
+
+            NonceCall::<T>::insert(sindex, nonce + 1);
+			Ok(())
 		}
 
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn approve_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			call_index: CallIndex,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+            if Self::users_votes((sindex, call_index), who.clone()) {
+                return Err(Error::<T>::AlreadyVoted.into())
+            }
+            if Self::calls(sindex, call_index).is_none() {
+                return Err(Error::<T>::CallNotFound.into())
+            }
+
+            UsersVotes::<T>::insert((sindex, call_index), who.clone(), true);
+            Votes::<T>::insert(sindex, call_index, Self::votes(sindex, call_index) + 1);
+
+            Self::deposit_event(Event::<T>::CallVoted(supersig_id, call_index, who));
+
+            let threshold = Self::supersigs(sindex).unwrap().threshold;
+            let total_votes = Self::votes(sindex, call_index);
+
+            if total_votes >= threshold {
+                Self::execute_call(sindex, call_index);
+            }
+
+            Ok(())
+        }
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn remove_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			call_index: CallIndex,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+            if who != supersig_id {
+                return Err(Error::<T>::NotAllowed.into());
+            }
```

Comment:
I don't think the caller should be `supersig_id` but rather `preimage.provider`
Sound silly to go through one more shared call to remove a previous one.
Unless it take multiple calls to remove. 
Maybe both can coexist